File: javascript/test_helper.js
const assertEqual = (actual, expected) => {
  const actualJson = JSON.stringify(actual);
  const expectedJson = JSON.stringify(expected);
  if (actualJson !== expectedJson) {
    throw new Error(`Test failed!\nExpected: ${expectedJson}\nActual: ${actualJson}`);
  }
};

const runTests = (...tests) => {
  let failures = 0;

  tests.forEach(test => {
    console.log(`-- Running ${test.name}...`);
    try {
      test();
    } catch (e) {
      failures++;
      console.log(e, "\n");
    }
  });

  console.log(`Ran ${tests.length} tests, ${failures} failures.`);
  if (failures === 0) {
    console.log("All tests passed!");
  }
};

export { assertEqual, runTests };

File: ruby/test_helper.rb
module TestHelper
  def assert_equal(actual, expected)
    if expected != actual
      raise "Test failed!\nExpected: #{expected.inspect}\nActual: #{actual.inspect}"
    end
  end

  def run_tests(*tests)
    failures = 0

    tests.each do |test|
      puts "-- Running #{test.name}..."
      begin
        test.call
      rescue StandardError => e
        failures += 1
        puts e, "\n"
      end
    end

    puts "Ran #{tests.length} tests, #{failures} failures."
    if failures.zero?
      puts "All tests passed!"
    end

  end
end
